src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -426,6 +432,26 @@ public String actualPathForPublicPath(String publicPath) {
         return null;
     }
 
+    /**
+     * Checks whether a path (defined by a URI) contains a certain sub-path (defined by a URI)).
+     *
+     * @param path    URI to be checked
+     * @param subPath URI to check
+     * @throws IllegalArgumentException if the provided path does not contain the provided sub-path
+     */
+    public static void sanitizeByCheckingIfPathContainsSubPathElseThrow(URI path, URI subPath) {
+        // Only act if path and subpath exist and if neither of them points to a temp location
+        if (path != null && !path.getPath().contains("files/temp") && subPath != null && !subPath.getPath().contains("files/temp")) {
```

Comment:
Doesn't that mean if an URI were to contain `files/temp` a file traversal would still be possible?

Why not normalise temporary paths too? I don't see why this should be any different.

If it starts with `api/files/temp` we should make sure that we also end up in that score.

Keyword: Safe default!

Reply:
Addressed by [57ebbd70da44d589f7c61fa9f42a7d4742ab1d71](https://github.com/ls1intum/Artemis/pull/7170/commits/57ebbd70da44d589f7c61fa9f42a7d4742ab1d71).